packages/controllers/src/snaps/SnapController.ts

```diff
@@ -1027,27 +1027,51 @@ export class SnapController extends BaseController<
    * Results from this method should be efficiently serializable.
    *
    * @param origin - The origin requesting the snap.
-   * @param snapId - The id of the snap.
-   * @param version - The version of the snap to install.
+   * @param _snapId - The id of the snap.
+   * @param versionRange - The semver range of the snap to install.
    * @returns The resulting snap object, or an error if something went wrong.
    */
   private async processRequestedSnap(
     origin: string,
-    snapId: SnapId,
-    version: string,
+    _snapId: SnapId,
+    versionRange: string,
   ): Promise<ProcessSnapResult> {
```

Comment:
I wonder if there is a way to simplify this function. with these additions there are now two paths that both call `authorize` etc. Do you think it's worth it to combine it somehow?

Reply:
As in `updateSnap` calls it and the remaining parts of this function calls it too

Reply:
Also does `startSnap` etc

Reply:
I looked at it, and couldn't find any nice way to abstract it out cleanly.